### PR TITLE
[FEATURE][S/C] Expose isServer property

### PR DESCRIPTION
**Pull request by Drakulix**
Adds a way for clients to tell if another participant of the session is a server.
Useful for GUI integration.

### CODE REVIEW

**Comment by srossbach** [comment]
Can you give a specific use case / user story that makes this feature necessary ? My guess is that you do not want to display the host/server in the session overview ?

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java [hunk 1]
```diff
@@ -34,19 +34,22 @@ public void addUser(User user, long flags) {
 
         public long flags;
         public JID jid;
+        public boolean isServer;
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make isServer a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMul&open=AWarMreBq3jbRxE0bMul)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMul

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java [hunk 2]
```diff
@@ -34,19 +34,22 @@ public void addUser(User user, long flags) {
...
         public int colorID;
         public int favoriteColorID;
         public Permission permission;
 
         private static UserListEntry create(User user, long flags) {
-            return new UserListEntry(user.getJID(), user.getColorID(),
-                user.getFavoriteColorID(), user.getPermission(), flags);
+            return new UserListEntry(user.getJID(), user.isServer(),
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMup&open=AWarMreBq3jbRxE0bMup)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMup

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java [hunk 3]
```diff
@@ -34,19 +34,22 @@ public void addUser(User user, long flags) {
 
         public long flags;
         public JID jid;
+        public boolean isServer;
         public int colorID;
         public int favoriteColorID;
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make favoriteColorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMun&open=AWarMreBq3jbRxE0bMun)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMun

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java [hunk 4]
```diff
@@ -34,19 +34,22 @@ public void addUser(User user, long flags) {
 
         public long flags;
         public JID jid;
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make jid a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreAq3jbRxE0bMuk&open=AWarMreAq3jbRxE0bMuk)

[ISSUE_KEY]: AWarMreAq3jbRxE0bMuk

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java [hunk 5]
```diff
@@ -34,19 +34,22 @@ public void addUser(User user, long flags) {
...
+        public boolean isServer;
         public int colorID;
         public int favoriteColorID;
         public Permission permission;
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make permission a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMuo&open=AWarMreBq3jbRxE0bMuo)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMuo

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java [hunk 6]
```diff
@@ -34,19 +34,22 @@ public void addUser(User user, long flags) {
 
         public long flags;
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make flags a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreAq3jbRxE0bMuj&open=AWarMreAq3jbRxE0bMuj)

[ISSUE_KEY]: AWarMreAq3jbRxE0bMuj

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java [hunk 7]
```diff
@@ -34,19 +34,22 @@ public void addUser(User user, long flags) {
...
         public JID jid;
+        public boolean isServer;
         public int colorID;
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make colorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMum&open=AWarMreBq3jbRxE0bMum)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMum

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java [hunk 8]
```diff
@@ -1107,7 +1109,8 @@ private SarosSession(final String id, IContainerContext context,
                 .getInt(ColorNegotiationHook.KEY_INITIAL_COLOR);
             int hostFavoriteColorID = hostProperties
                 .getInt(ColorNegotiationHook.KEY_FAV_COLOR);
-            hostUser = new User(host, true, false, hostColorID,
+            boolean hostIsServer = hostProperties.getBoolean(ServerStatusHook.KEY_IS_SERVER);
+            hostUser = new User(host, true, false, hostIsServer, hostColorID,
                 hostFavoriteColorID);
             hostUser.setInSession(true);
             participants.put(hostUser.getJID(), hostUser);
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMrdiq3jbRxE0bMuX&open=AWarMrdiq3jbRxE0bMuX)

[ISSUE_KEY]: AWarMrdiq3jbRxE0bMuX

**Comment by Drakulix** [comment]
> Isn't the server property also announced as XMPP feature ? So basically you already know that the given JID is a "server" before the session starts at all.

Yes you are right, it is. However I am not sure, if that is the right way to go. Being a server isn't really a feature you need to query, but rather describing the lack of features.

Also I would prefer not using the XMPP stack for this.
[`isFeatureSupported`](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/net/xmpp/discovery/DiscoveryManager.java#L218) might return `null` and there is no good way to handle this case in my opinion. This method is reliable.

**Comment by srossbach** [comment]
I know that this XMPP stuff is not the right way to archive this. But if you look at this solution. You modified over 10 files and added multiple LOCs to just expose a single boolean value. Furthermore there is now a flag in the core that is associated with a non core system, the server. Also it is now confusing that we have a host, a server, and a client property.

**Comment by Drakulix** [comment]
> I know that this XMPP stuff is not the right way to archive this. But if you look at this solution. You modified over 10 files and added multiple LOCs to just expose a single boolean value. Furthermore there is now a flag in the core that is associated with a non core system, the server. Also it is now confusing that we have a host, a server, and a client property.

- Well yes, because there is a lot to it. If you think it is too confusing, we should think about naming things differently instead.

- I know this adds some lines, but it is essentially the same code as for sharing the colors or any other value over the new hook system. If you think this should be redesigned again, feel free to open an issue and assign me. But that is the way data exchange during negotiation works now or did I miss something?

- I know this adds to core, but I do not see a way to add this to the User class any other way. Also there might (technically) be other headless implementations of saros then just the server. Maybe I should rename the property to "isHeadless" instead? Would that fit better into core?

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMup&open=AWarMreBq3jbRxE0bMup)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMup

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make isServer a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMul&open=AWarMreBq3jbRxE0bMul)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMul

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 7 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make colorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSN1&open=AWbtivsTW3Y69bz5JSN1)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSN1

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 4 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make jid a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSN0&open=AWbtivsTW3Y69bz5JSN0)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSN0

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 6 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make flags a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSNz&open=AWbtivsTW3Y69bz5JSNz)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSNz

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java (hunk 8 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivrnW3Y69bz5JSKM&open=AWbtivrnW3Y69bz5JSKM)

[ISSUE_KEY]: AWbtivrnW3Y69bz5JSKM

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 3 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make favoriteColorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSN2&open=AWbtivsTW3Y69bz5JSN2)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSN2

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 5 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make permission a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsUW3Y69bz5JSN4&open=AWbtivsUW3Y69bz5JSN4)

[ISSUE_KEY]: AWbtivsUW3Y69bz5JSN4

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make isServer a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMul&open=AWarMreBq3jbRxE0bMul)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMul

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMup&open=AWarMreBq3jbRxE0bMup)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMup

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 5 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make permission a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsUW3Y69bz5JSN4&open=AWbtivsUW3Y69bz5JSN4)

[ISSUE_KEY]: AWbtivsUW3Y69bz5JSN4

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make isServer a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMul&open=AWarMreBq3jbRxE0bMul)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMul

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMup&open=AWarMreBq3jbRxE0bMup)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMup

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 3 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make favoriteColorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSN2&open=AWbtivsTW3Y69bz5JSN2)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSN2

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 7 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make colorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSN1&open=AWbtivsTW3Y69bz5JSN1)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSN1

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java (hunk 8 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivrnW3Y69bz5JSKM&open=AWbtivrnW3Y69bz5JSKM)

[ISSUE_KEY]: AWbtivrnW3Y69bz5JSKM

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 6 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make flags a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSNz&open=AWbtivsTW3Y69bz5JSNz)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSNz

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 4 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make jid a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWbtivsTW3Y69bz5JSN0&open=AWbtivsTW3Y69bz5JSN0)

[ISSUE_KEY]: AWbtivsTW3Y69bz5JSN0

**Comment by Drakulix** [comment]
> IIRC the result of your hook goes into the session properties. So what is the point of adding this property to the User class ? It is a property that will never change during a session. There is no need to broadcast this using the UserEntry extension.

Yes there is a need for implementing a consistent UI. Properties are not broadcasted between clients, so if at any point in the future there may be more then one headless client and/or that headless client is not guaranteed to be the host, there is no way to tell for every client, if any other session participant is headless.

Right now this is indeed useless, because only the host can ever be a server and servers will not join each other. So if you prefer I can remove that code, but I would rather not.

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMup&open=AWarMreBq3jbRxE0bMup)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMup

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make isServer a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMul&open=AWarMreBq3jbRxE0bMul)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMul

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 3 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make favoriteColorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWc2MMo10EcSutaxS9dn&open=AWc2MMo10EcSutaxS9dn)

[ISSUE_KEY]: AWc2MMo10EcSutaxS9dn

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 6 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make flags a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWc2MMo10EcSutaxS9dk&open=AWc2MMo10EcSutaxS9dk)

[ISSUE_KEY]: AWc2MMo10EcSutaxS9dk

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 4 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make jid a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWc2MMo10EcSutaxS9dl&open=AWc2MMo10EcSutaxS9dl)

[ISSUE_KEY]: AWc2MMo10EcSutaxS9dl

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make isServer a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMul&open=AWarMreBq3jbRxE0bMul)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMul

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java (hunk 8 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWc2MMoQ0EcSutaxS9dZ&open=AWc2MMoQ0EcSutaxS9dZ)

[ISSUE_KEY]: AWc2MMoQ0EcSutaxS9dZ

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Code Smell Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWarMreBq3jbRxE0bMup&open=AWarMreBq3jbRxE0bMup)

[ISSUE_KEY]: AWarMreBq3jbRxE0bMup

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 5 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make permission a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWc2MMo10EcSutaxS9do&open=AWc2MMo10EcSutaxS9do)

[ISSUE_KEY]: AWc2MMo10EcSutaxS9do

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java (hunk 7 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/communication/extensions/UserListExtension.java]: Vulnerability Vulnerability: Make colorID a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=284&issues=AWc2MMo10EcSutaxS9dm&open=AWc2MMo10EcSutaxS9dm)

[ISSUE_KEY]: AWc2MMo10EcSutaxS9dm

**Pull request closed without merging**